Merging actions/stale:master -> master

Chocrates opened the pull request:
There were merge conflicts in the compile index.js so manually pulling the changes in.

Getting this ready to upstream (or at least start the conversation with the maintainers)

hross committed:
Fix issue close/stale dates (#72)

Added tests to confirm behavior.

hross committed:
Fix an issue where the bot doesn't ignore its own comments (#73)

Commit:
Dropping dist/index.js from pr

zkoppert commented:
@Chocrates Any insight on the failing build?

Chocrates commented:
> @Chocrates Any insight on the failing build?

Nope, i told it to re-run but ill have to dig into it to see whats up when it fails again.

froi commented on __tests__/main.test.ts:
```diff
@@ -581,3 +581,94 @@ test('stale label should be removed if a comment was added to a stale issue', as
 });
+
+test('stale label should not be removed if a comment was added by the bot (and the issue should be closed)', async () => {
+  github.context.actor = 'abot';
```
```suggestion
  octomock.context.actor = 'abot';
```

The `github` object was replaced with `octomock`. Committing this suggestion will cause another error, the `context` object field is __undefined__ at this point.

froi reviewed (changes requested):
We need to change the use of the `github` object to use `octomock`. Changing it wont fix the issue though, the `context` object needs to be created correctly to have it pass.

I'm looking at the issue and taking a look at [octomock](https://github.com/chocrates/octomock) to see what options we have. @Chocrates any thoughts?

Commit:
Update __tests__/main.test.ts

Co-authored-by: Froilan Irizarry Rivera <1918027+froi@users.noreply.github.com>

froi commented:
@Chocrates I found the issue with this PR. The tests still need the use of `@actions/github`, `octomock` is not a total replacement.

The `@actions/github` import provides a __github__ object that contains a `github.context` object that represents the context of the action itself. The way Node.js works with import keeps the `github` object in the Node cache once it's imported in `__test__/main.test.ts`. 

Setting `github.context.actor = 'abot'` in `__test__/main.test.ts` carries over to `IssueProcessor.isIssueStillStale` and the validation in the return works properly

```js
return (
      comments.filter(
        comment =>
          comment.user.type === 'User' &&
          comment.user.login !== github.context.actor // This now works!
      ).length > 0
```

My initial suggestion needs to be reverted and `import {GitHub} from '@actions/github';` needs to be changed to `import * as github from '@actions/github';`.

froi committed:
Revest PR code suggestion

Reverted a code suggestion made in PR #4 to change github.context.actor to octomock.context.actor. This needed to be reverted to keep the github object with the correct context in the test as this represents the action's context and is part of what we are testing.
Changed the github import to be * as github to satisfy the needs of tests and provide the mocked action context.

Signed-off-by: Froilan Irizarry Rivera <1918027+froi@users.noreply.github.com>

froi commented:
@Chocrates I've pushed the fixes to the test in https://github.com/ActionsDesk/stale/pull/4/commits/e6b78412ac7d42b22812e85cb1be4259bc9be79f.

Checks are now green.

froi reviewed (approved):
Checks are now green. See my comments above and commit message here https://github.com/ActionsDesk/stale/pull/4/commits/e6b78412ac7d42b22812e85cb1be4259bc9be79f for the 411

:shipit: if you are satisfied.

Chocrates reviewed (commented):
:ship:

Chocrates commented:
Thanks for digging in to this @froi
